src/diff/props.js

```diff
@@ -76,7 +76,7 @@ function setProperty(dom, name, value, oldValue, isSvg) {
 	else if (name[0]==='o' && name[1]==='n') {
 		let useCapture = name !== (name=name.replace(/Capture$/, ''));
 		let nameLower = name.toLowerCase();
-		name = (nameLower in self ? nameLower : name).substring(2);
+		name = ((typeof self!=='undefined' && nameLower in self) ? nameLower : name).slice(2);
```

Comment:
Hmm is there a reason we do `nameLower in self` instead of `nameLower in dom`?

Reply:
dom does not contain certain events like for example the animation API these are defined on window but not on dom nodes.

Reply:
Hey @andrewiggins 👋 Yeah, so we should support the events from Global Event Interface(https://developer.mozilla.org/en-US/docs/Web/API/GlobalEventHandlers) and some of these events, like `ontransitionend`, are not on the `dom` instance.

This was added as part of https://github.com/developit/preact/pull/1590

Do you see any issues with this?

Reply:
what about `nameLower in dom.ownerDocument`?

Reply:
Right! I had forgotten about that. Thanks

Reply:
Using @robertknight suggestion `dom.ownerDocument.defaultView` is actually what were looking for? Since `dom.ownerDocument === document` and these Global Events are not defined on `document` but rather on `window`. 

This will actually turn-up smaller in size by just adding `+7B` to core